rfc/rfc-0000-add-template-field-to-runnable.md

```diff
@@ -0,0 +1,195 @@
+  name: linter
+spec:
+  runTemplateRef:
+    name: md-linting-pipelinerun
+  inputs:
+    repository: some-repo
+    revision: some-sha
+  serviceAccountName: pipeline-run-management-sa
+```
+
+This would become the single Runnable here:
+
+```yaml
+apiVersion: carto.run/v1alpha1
+kind: Runnable
+metadata:
+  name: linter
+spec:
+  serviceAccountName: pipeline-run-management-sa
+  template:
+    apiVersion: tekton.dev/v1beta1
+    kind: PipelineRun
+    metadata:
+      generateName: some-name-pipeline-run-
+    spec:
+      pipelineRef:
+        name: linter-pipeline
+      params:
+        - name: repository
+          value: some-repo
+        - name: revision
+          value: some-sha
+      workspaces:
+        - name: shared-workspace
+          volumeClaimTemplate:
+            spec:
+              accessModes:
+                - ReadWriteOnce
+              resources:
+                requests:
+                  storage: 256Mi
+  outputs:
+    url: spec.params[?(@.name=="repository")].value
+    revision: spec.params[?(@.name=="revision")].value
+```
+
+The target persona is Cartographer maintainers; the main driver of this is to enable app operators to create templates
+that leverage Runnable without needing to submit/manage ClusterRunTemplates.
+See [this RFC](https://github.com/vmware-tanzu/cartographer/pull/705) for discussion of that use case.
+
+# How it Works
+
+[how-it-works]: #how-it-works
+
+The object defined in the Runnable is submitted to the cluster. The runnable exposes the output fields defined in
+the runnable's own status. Care should be taken to garbage collect objects submitted by a runnable if the GVK or
+name of the object is changed.
+
+# Migration
+
+[migration]: #migration
+
+See [spec-changes].
+
+This RFC would require a version bump for Runnable. There is no technical reason that Runnables with valid
+references to ClusterRunTemplate could not be translated into Runnables with template fields embedded.
+
+The [alternatives] section describes keeping `runTemplateRef` alongside `template`. This option presents a purely
+additive change.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+This runs a bit counter to the design of Cartographer, which has template objects that pair with an owner that
+provides values to the template. These template objects are meant to be reuseable. In this case they are not.
+
+This could be seen as the tail wagging the dog. Rather than simplifying the direct use case of Runnable, this makes
+the direct use case worse, in pursuit of slightly smoothing out the automation of another feature (supply chain
+templates automatically creating Runnables).
+
+# Alternatives
+
+[alternatives]: #alternatives
+
+Rather than removing ClusterRunTemplates altogether, the `template` field option could exist alongside the the
+`runTemplateRef` field. This has the advantage of preserving the owner-template relationship described in the
+[drawbacks] section above. This does lead to its own complications:
```

Comment:
Preserving backwards compatibility is important. It's fine to deprecate the old behavior if the eventual goal is to remove it. Having a window where the old and new models both work gives users time to migrate.